src/common/settings.ts

```diff
@@ -196,16 +196,36 @@ function getOptionalGlobalValue<T>(config: WorkspaceConfiguration, key: string):
   return inspect?.globalValue;
 }
 
+/**
+ * Returns the global settings for the extension.
+ *
+ * ## Notes
+ *
+ * The global settings do not belong to a specific workspace. This means that
+ * variables such as `${workspaceFolder}` or `${workspaceFolder:...}` are not
+ * resolved. The language server does not support these variables, so they are
+ * filtered out. For example, if `configuration` has either of these variables,
+ * it will be set to `null`.
+ */
 export async function getGlobalSettings(namespace: string): Promise<ISettings> {
   const config = getConfiguration(namespace);
+
+  let configuration = getGlobalValue<string | object | null>(config, "configuration", null);
+  if (typeof configuration === "string" && configuration.search(/\$\{workspaceFolder/) !== -1) {
+    logger.info(
```

Comment:
Should we log this with `warn` or even error?

Reply:
I don't think so, it's not a user error but just something to note of. Refer to my other comment.